examples/misc/cursor_demo_sgskip.py

```diff
@@ -22,70 +28,154 @@
 
 
 class Cursor:
+    """
+    A cross hair cursor.
+    """
     def __init__(self, ax):
         self.ax = ax
-        self.lx = ax.axhline(color='k')  # the horiz line
-        self.ly = ax.axvline(color='k')  # the vert line
-
-        # text location in axes coords
-        self.txt = ax.text(0.7, 0.9, '', transform=ax.transAxes)
+        self.horizontal_line = ax.axhline(color='k', lw=0.8, ls='--')
+        self.vertical_line = ax.axvline(color='k', lw=0.8, ls='--')
+        # text location in axes coordinates
+        self.text = ax.text(0.72, 0.9, '', transform=ax.transAxes)
 
     def mouse_move(self, event):
         if not event.inaxes:
             return
-
         x, y = event.xdata, event.ydata
         # update the line positions
-        self.lx.set_ydata(y)
-        self.ly.set_xdata(x)
-
-        self.txt.set_text('x=%1.2f, y=%1.2f' % (x, y))
+        self.horizontal_line.set_ydata(y)
+        self.vertical_line.set_xdata(x)
+        self.text.set_text('x=%1.2f, y=%1.2f' % (x, y))
         self.ax.figure.canvas.draw()
 
 
-class SnaptoCursor:
+x = np.arange(0, 1, 0.01)
+y = np.sin(2 * 2 * np.pi * x)
+
+fig, ax = plt.subplots()
+ax.set_title('Simple cursor')
+ax.plot(x, y, 'o')
+cursor = Cursor(ax)
+fig.canvas.mpl_connect('motion_notify_event', cursor.mouse_move)
+
+
+##############################################################################
+# Faster redrawing using blitting
+# """""""""""""""""""""""""""""""
+# This technique stores the rendered plot as a background image. Only the
+# changed artists (cross hair lines and text) are rendered anew. They are
+# combined with the background using blitting.
+#
+# This technique is significantly faster. It requires a bit more setup because
+# the background has to be stored without the cross hair lines (see
+# ``create_background()``). Additionally, a new background has to be created
+# whenever the figure changes. The example implements this for resizing. If
```

Comment:
This also doesn't handle pan & zoom, which are enabled by default.

Reply:
I know. Thought I could get away with that.

I've now changed to recreating the background on every draw. This is a bit more fiddly but covers all possible changes.